packages/next-swc/crates/next-core/js/src/internal/shims-client.ts

```diff
@@ -17,6 +17,3 @@ globalThis.__webpack_require__ = (name) => {
 
 // initialize() needs `__webpack_public_path__` to be defined.
 globalThis.__webpack_public_path__ = undefined
-
-// Avoids Next loading _next/static/[buildId]/_devMiddlewareManifest.json
-globalThis.__DEV_MIDDLEWARE_MATCHERS = []
```

Comment:
If this is defined, then we'll never fetch the middleware matchers from the server. The fetcher will cache the fetched value afterwards.